packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,390 @@
+   * The VirtualService this GatewayRoute directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
+}
+
+/**
+ * Properties specific for a GRPC GatewayRoute
+ */
+export interface GrpcRouteSpecProps {
+  /**
+   * The criterion for determining a request match for this GatewayRoute
+   */
+  readonly match: GrpcGatewayRouteMatch;
+  /**
+   * The VirtualService this GatewayRoute directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
+}
+
+/**
+ * All Properties for GatewayRoute Specs
+ */
+export interface GatewayRouteSpecConfig {
+  /**
+   * The spec for an http gateway route
+   *
+   * @default - no http spec
+   */
+  readonly httpSpecConfig?: CfnGatewayRoute.HttpGatewayRouteProperty;
+  /**
+   * The spec for an http2 gateway route
+   *
+   * @default - no http2 spec
+   */
+  readonly http2SpecConfig?: CfnGatewayRoute.HttpGatewayRouteProperty;
+  /**
+   * The spec for a grpc gateway route
+   *
+   * @default - no grpc spec
+   */
+  readonly grpcSpecConfig?: CfnGatewayRoute.GrpcGatewayRouteProperty;
+}
+
+/**
+ * Used to generate specs with different protocols for a GatewayRoute
+ */
+export abstract class GatewayRouteSpec {
+  /**
+   * Creates an HTTP Based GatewayRoute
+   *
+   * @param props - no http gateway route
+   */
+  public static httpRouteSpec(props: HttpRouteSpecProps): GatewayRouteSpec {
+    return new HttpGatewayRouteSpec(props, Protocol.HTTP);
+  }
+
+  /**
+   * Creates an HTTP2 Based GatewayRoute
+   *
+   * @param props - no http2 gateway route
+   */
+  public static http2RouteSpec(props: HttpRouteSpecProps): GatewayRouteSpec {
+    return new HttpGatewayRouteSpec(props, Protocol.HTTP2);
+  }
+
+  /**
+   * Creates an GRPC Based GatewayRoute
+   *
+   * @param props - no grpc gateway route
+   */
+  public static grpcRouteSpec(props: GrpcRouteSpecProps): GatewayRouteSpec {
+    return new GrpcGatewayRouteSpec(props);
+  }
+
+  /**
+   * Called when the GatewayRouteSpec type is initialized. Can be used to enforce
+   * mutual exclusivity with future properties
+   */
+  public abstract bind(scope: cdk.Construct): GatewayRouteSpecConfig;
+}
+
+class HttpGatewayRouteSpec extends GatewayRouteSpec {
+  /**
+   * The criterion for determining a request match for this GatewayRoute.
+   *
+   * @default - matches on '/'
+   */
+  readonly match?: HttpGatewayRouteMatch;
+  /**
+   * The VirtualService this GatewayRoute directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
+
+  /**
+   * Type of route you are creating
+   */
+  readonly routeType: Protocol;
+
+  constructor(props: HttpRouteSpecProps, protocol: Protocol.HTTP | Protocol.HTTP2) {
+    super();
+    this.routeTarget = props.routeTarget;
+    this.routeType = protocol;
+    this.match = props.match;
+  }
+  public bind(_scope: cdk.Construct): GatewayRouteSpecConfig {
+    const prefixPath = this.match ? this.match.prefixPath : '/';
+    if (prefixPath[0] != '/') {
+      throw new Error('Prefix Path must start with \'/\'');
```

Comment:
It's good practice to include the value that's causing the error in your exceptions - this way, it's much easier to find the reason a validation fails:

```suggestion
      throw new Error(`Prefix path must start with '/', got: ${prefixPath}`);